### PR TITLE
Rename `NavigationLink` component to `NavigationAction`

### PR DESCRIPTION
**Pull request by crishpeen**

### CODE REVIEW

File: packages/web-react/src/components/Navigation/README.md [hunk 1]
```diff
@@ -6,7 +6,7 @@ It consists of a these parts:
 
 - [Navigation](#navigation)
 - [NavigationItem](#navigation-item)
-- [NavigationLink](#navigation-link)
+- [NavigationAction](#navigation-link)
```
Comment (adamkudrna) [inline packages/web-react/src/components/Navigation/README.md]: Maybe?

- [Navigation](#navigation)
   - [NavigationItem](#navigation-item)
      - [NavigationAction](#navigation-link)

File: packages/web-react/src/components/Navigation/__tests__/useNavigationActionProps.test.ts [hunk 2]
```diff
@@ -77,17 +77,17 @@ describe('useNavigationLinkProps', () => {
   });
 
   it('should return for elementType', () => {
-    const props: SpiritNavigationLinkProps<'div'> = {
+    const props: SpiritNavigationActionProps<'div'> = {
       elementType: 'div',
```
Comment (adamkudrna) [inline packages/web-react/src/components/Navigation/__tests__/useNavigationActionProps.test.ts:81]: Off topic: I'd prefer testing the real-world `<button>` type here.

File: packages/web-react/src/components/UNSTABLE_Header/README.md [hunk 3]
```diff
@@ -125,7 +125,7 @@ If you need the whole Header fluid you can do it by adding the `isFluid` prop to
 
 You can use the [`Navigation`][web-react-navigation] component inside the `UNSTABLE_Header` component.
 
-The `NavigationLink` components will inherit the `UNSTABLE_Header` height and set the navigation
+The `NavigationAction` components will inherit the `UNSTABLE_Header` height and set the navigation
 link height to the same value.
```
Comment (adamkudrna) [inline packages/web-react/src/components/UNSTABLE_Header/README.md]: "link"

File: packages/web-react/src/components/UNSTABLE_Header/demo/HeaderWithNavigation.tsx [hunk 4]
```diff
@@ -4,7 +4,7 @@ import { ButtonLink } from '../../ButtonLink';
 import { Container } from '../../Container';
 import { Flex } from '../../Flex';
 import { Icon } from '../../Icon';
-import { Navigation, NavigationItem, NavigationLink } from '../../Navigation';
+import { Navigation, NavigationItem, NavigationAction } from '../../Navigation';
```
Comment (adamkudrna) [inline packages/web-react/src/components/UNSTABLE_Header/demo/HeaderWithNavigation.tsx:7]: 🔤

File: packages/web/src/scss/components/Navigation/README.md [hunk 5]
```diff
@@ -1,71 +1,71 @@
...
 
-It centers its children vertically and if the children are not NavigationLink components,
+It centers its children vertically and if the children are not NavigationAction components,
 it will insert a gap between them.
 
-ℹ️ If you plan to provide a list of links, wrap them in a `ul` and `li` elements. If not, you can use the
+ℹ️ If you plan to provide a list of actions, wrap them in a `ul` and `li` elements. If not, you can use the
 `nav` element directly.
 
 ℹ️ Don't forget to add the `aria-label` attribute to the `Navigation` component for correct accessible state.
```
Comment (adamkudrna) [inline packages/web/src/scss/components/Navigation/README.md:30]:
```suggestion
ℹ️ Don't forget to add the `aria-label` attribute to the `Navigation` component for correct accessible title.
```

File: packages/web/src/scss/components/UNSTABLE_Header/README.md [hunk 6]
```diff
@@ -121,8 +121,8 @@ If you need the whole Header fluid you can do it by adding the `Container--fluid
 
 You can use the [`Navigation`][web-navigation] component inside the `UNSTABLE_Header` component.
 
-The `NavigationLink` components will inherit the `UNSTABLE_Header` height and set the navigation
-link height to the same value.
+The `NavigationAction` components will inherit the `UNSTABLE_Header` height and set the navigation
```
Comment (adamkudrna) [inline packages/web/src/scss/components/UNSTABLE_Header/README.md:124]: This is the other way round now.

**Review by adamkudrna — APPROVED** [review]

File: packages/web-react/src/components/Navigation/__tests__/useNavigationActionProps.test.ts (hunk 2 above)
Comment (crishpeen) [reply packages/web-react/src/components/Navigation/__tests__/useNavigationActionProps.test.ts:81]: ok, I will add it when I introduce button styles to navigationaction

**Review by literat — APPROVED** [review]

**Review by pavelklibani — APPROVED** [review]

**Pull request merged**